### DISCUSSION TITLE
Consider consolidating GitHub wiki actions?

### DISCUSSION THREAD

**Discussion by jcbhmr** [spenserblack/actions-wiki#78] [General]
@spenserblack would you be open to revisiting the idea of combining github wiki actions? Particularily @Andrew-Chen-Wang's Andrew-Chen-Wang/github-wiki-action which I have contributed to recently.

I think the goals of this project align _reasonably well_ with Andrew-Chen-Wang/github-wiki-action, particularly given the recent shift from "I dream of lots of features on by default" to a more restrained "just push the folder to the wiki" action with a recommendation to use _other_ actions to do complex stuff.

previously: https://github.com/Andrew-Chen-Wang/github-wiki-action/discussions/26 and https://github.com/Andrew-Chen-Wang/github-wiki-action/issues/8

now: https://github.com/Andrew-Chen-Wang/github-wiki-action/issues/8#issuecomment-1550311999 and https://github.com/Andrew-Chen-Wang/github-wiki-action#preprocessing

On a practical note, I have contributed a few features that I've proposed here to Andrew-Chen-Wang/github-wiki-action and they worked out pretty well! I think it'd be foolish to re-implement them a second time (or copy-paste 🤣) vs nudging towards Andrew-Chen-Wang/github-wiki-action

- github-server-url option
- using $GIT_DIR and $GIT_WORK_TREE
- providing an alternative strategy to push to the wiki
- using `gh auth setup-git` properly to configure auto-authentication with github push/clone urls

more info on the recent v4 stuff: https://github.com/Andrew-Chen-Wang/github-wiki-action/issues/30

**TLDR:** spenserblack/actions-wiki and Andrew-Chen-Wang/github-wiki-action seem to fill the same usecase and niche now. Can nudge spenserblack/actions-wiki towards Andrew-Chen-Wang/github-wiki-action? Or is there a specific direction or usecase that we'd like to build on for spenserblack/actions-wiki?

**Comment by spenserblack** [comment]
I probably won't abandon/archive this project in favor of github-wiki-action, since I initially made it to be simply for personal use. However, I understand not wanting to duplicate work, so if you decide to make github-wiki-action the primary project that you contribute to, that's fine with me :slightly_smiling_face: 

Feel free to make a PR referencing this somehow. Right now github-wiki-action is referred to as "a direct competitor," but you could possibly reword that to something like "receives more active contribution from a major contributor to this project."

**Reply by jcbhmr** [reply]
If you have a vision on a particular direction (ex: a TS-based github wiki action) then that's a great distinguishing factor! Otherwise yeah it's a direct feature-inferior competitor.

The thing I wanted to bring up here is that if I merge my changes, then this spenserblack/actions-wiki and the Andrew-Chen-Wang/github-wiki-action will essentially be the same exact action 😜

**Reply by jcbhmr** [reply]
You know... You could try to push for a TS rewrite of Andrew-Chen-Wang/github-wiki-action 😂 I do have write/merge permissions if you're worried about timely-ness. And, I'm sure Andrew-Chen-Wang would love to have another contributor improve the project!

Basically what I'm getting at is "why are there two projects that do _literally the same thing_ (or close enough to the same thing) and is there a way to maybe merge the two?"

**Reply by jcbhmr** [reply]
from #80 

> I find TS easier to contribute to, and also I think it's easier to write tests. That's my basic reasoning. Also, I think the language looks prettier 🙃

I know I'm beating a dead horse here but... Would you consider consolidating GitHub wiki Actions and rewriting https://github.com/Andrew-Chen-Wang/github-wiki-action in TS instead? 🤪 I want to avoid splintering and fracturing. ❤️ If we can make one popular thing better that's a very positive impact. 🥇

**Reply by spenserblack** [reply]
If there's any interest in rewriting github-wiki-action in TS I'd be happy to help, but I honestly don't see any gain by consolidating. It's kind of the nature of Open Source that clones of projects will pop up, and I think it's healthy to have varying implementations of the same solution.

**Reply by jcbhmr** [reply]
> I think it's healthy to have varying implementations of the same solution.

While I agree _in principle_ that competition to foster improvement is good, I also see value in the Python wisdom:

> There should be one-- and preferably only one --obvious way to do it.
https://peps.python.org/pep-0020/

In this case, I think trying to **improve the original (or popular) thing** is better than making a clone. But that's me.

**Reply by spenserblack** [reply]
How many Python package managers are there? :laughing:

**Reply by Andrew-Chen-Wang** [reply]
@jcbhmr I don't understand the meaning of competition in the context of OSS. If there is maintenance for an existing solution, then that's all that's needed for OSS. Maintenance is key. Please don't bother other maintainers unless they've abandoned a project that has popularity (ie needs a new maintainer).

**Reply by jcbhmr** [reply]
> Please don't bother other maintainers unless they've abandoned a project that has popularity (ie needs a new maintainer).

Sounds good, I'll drop the subject. 👍

**Reply by spenserblack** [reply]
For what it's worth, I'd welcome a PR that updates the README to recommend @Andrew-Chen-Wang's project as a more mature or feature-rich alternative. @Andrew-Chen-Wang does make a good point about the usage of the word "competition," so at the very least the README probably shouldn't call their project a "direct competitor."

**Comment by jcbhmr** [comment]
Opened a PR #82
